home/tests/test_display_question_feature.py

```diff
@@ -107,10 +91,97 @@ def test_response_context(self, response):
+            answerB = '(<Answer: Answer B>, \'bi bi-hand-thumbs-up\', \'bi bi-hand-thumbs-down-fill'
+            assert answerB in str(response.context['answers_structure'])
+
+        @pytest.mark.django_db
+        def test_logged_user_unlike(self, client):
+            """
+            in this test, a logged user press like on (answer B) which he liked before
+            and make it un-liked
+            """
+            user = Profile.objects.all().get(user_id=1).user
+            client.force_login(user)
+            url = reverse('thumbs')
+            response = client.post(url, {'button': "like 10"})
+            response = client.get(response.url)
+            answerB = '(<Answer: Answer B>, \'bi bi-hand-thumbs-up\', \'bi bi-hand-thumbs-down'
+            assert answerB in str(response.context['answers_structure'])
+
+        @pytest.mark.django_db
+        def test_logged_user_undislike(self, client):
+            """
+            in this test, a logged user press dislike on (answer A) which he disliked before
+            and make it un-disliked
+            """
+            user = Profile.objects.all().get(user_id=1).user
+            client.force_login(user)
+            url = reverse('thumbs')
+            response = client.post(url, {'button': "dislike 9"})
+            response = client.get(response.url)
+            answerA = '(<Answer: Answer A>, \'bi bi-hand-thumbs-up\', \'bi bi-hand-thumbs-down'
+            assert answerA in str(response.context['answers_structure'])
+
+        @pytest.mark.django_db
+        def test_invalid_thumbs_call_from_unsigned_user(self, client):
+            """
+            Unsigned user should not be able to like an answer
+            """
+            url = reverse('thumbs')
+            response = client.post(url, {'button': "like 9"})
+            assert response.status_code == 404
+
+        @pytest.mark.django_db
+        def test_invalid_thumbs_call_with_invalid_answer_id(self, client):
+            """
+            Testing return of 404 page when calling thumbs with non-existing answer id
+            """
+            user = Profile.objects.all().get(user_id=1).user
+            client.force_login(user)
+            url = reverse('thumbs')
+            thumbs_args = f'like {Answer.objects.count()+1}'
+            response = client.post(url, {'button': thumbs_args})
+            assert response.status_code == 404
+
+        @pytest.mark.django_db
+        def test_login_link_shown_for_unsigned_user(self, client):
+            """
+            Tests if the page loads the link "login to like" for unsigned users
+            """
+            url = reverse('question-detail', args=[14])
+            display_question_response = client.post(url)
+            char_content = display_question_response.content.decode(display_question_response.charset)
+            assert 'Login</a> to like</small>' in char_content
+            assert 'disabled' in char_content
```

Comment:
Great test coverage - like\unlike and dislike\undislike logged in users, disable content fo unsigned user and invalid inputs.